Stack/Opc.Ua.Core/Stack/Configuration/ApplicationConfiguration.cs

```diff
@@ -96,28 +97,19 @@ public partial class ApplicationConfiguration
         /// Gets the file that was used to load the configuration.
         /// </summary>
         /// <value>The source file path.</value>
-        public string SourceFilePath
-        {
-            get { return m_sourceFilePath; }
-        }
+        public string SourceFilePath => m_sourceFilePath; 
 
         /// <summary>
-        /// Get or set the certificate validator which is configured to use.
+        /// Gets the certificate validator which is configured to use.
         /// </summary>
-        public CertificateValidator CertificateValidator
-        {
-            get { return m_certificateValidator; }
-            set { m_certificateValidator = value; }
-        }
+        public CertificateValidator CertificateValidator => m_certificateValidator;
```

Comment:
CertificateValidator property should remain settable from outside.
In some cases the application implements a custom CertificateValidator, e.g. with a custom Validate() implementation, and needs to assign it to the stack.

Reply:
I made this mutable again, moved initialization back into the `Initialize` method, added a unit test ensuring that it can be set publicly, and kept the change in the `Validate` method so it doesn't re-assign after validate is called.